packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -611,9 +645,11 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
         this.trace(log => log('MonacoEditorModel.revert - exit'));
     }
 
-    createSnapshot(): object {
+    createSnapshot(this: IResolvedTextEditorModel): ITextSnapshot;
+    createSnapshot(this: ITextEditorModel): ITextSnapshot | null;
+    createSnapshot(): ITextSnapshot | null {
         return {
-            value: this.getText()
+            read: () => this.getText()
```

Comment:
This implementation will probably not work well with monaco: it's supposed to sequentially return chunks of manageable size.

Reply:
Good point. I did this as the shortest path to type compatibility, but I've modified it now to delegate to the underlying Monaco model's `createSnapshot` method.